iot-edge/EdgeSolution/modules/CSharpModule/Program.cs

```diff
@@ -0,0 +1,172 @@
+namespace CSharpModule
+{
+    using System;
+    using System.IO;
+    using System.Runtime.InteropServices;
+    using System.Runtime.Loader;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using System.Collections.Generic;     // For KeyValuePair<>
+    using Microsoft.Azure.Devices.Shared; // For TwinCollection
+    using Newtonsoft.Json;                // For JsonConvert
+    class Program
+    {
+        static int counter;
+        static int temperatureThreshold { get; set; } = 25;
+        class MessageBody
+        {
+            public Machine machine { get; set; }
+            public Ambient ambient { get; set; }
+            public string timeCreated { get; set; }
+        }
+        class Machine
+        {
+            public double temperature { get; set; }
+            public double pressure { get; set; }
+        }
+        class Ambient
+        {
+            public double temperature { get; set; }
+            public int humidity { get; set; }
+        }
+        static void Main(string[] args)
+        {
+            Init().Wait();
+
+            // Wait until the app unloads or is cancelled
+            var cts = new CancellationTokenSource();
+            AssemblyLoadContext.Default.Unloading += (ctx) => cts.Cancel();
+            Console.CancelKeyPress += (sender, cpe) => cts.Cancel();
+            WhenCancelled(cts.Token).Wait();
+        }
+
+        /// <summary>
+        /// Handles cleanup operations when app is cancelled or unloads
+        /// </summary>
+        public static Task WhenCancelled(CancellationToken cancellationToken)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+            cancellationToken.Register(s => ((TaskCompletionSource<bool>)s).SetResult(true), tcs);
+            return tcs.Task;
+        }
+
+        /// <summary>
+        /// Initializes the ModuleClient and sets up the callback to receive
+        /// messages containing temperature information
+        /// </summary>
+        static async Task Init()
+        {
+            AmqpTransportSettings amqpSetting = new AmqpTransportSettings(TransportType.Amqp_Tcp_Only);
+            ITransportSettings[] settings = { amqpSetting };
+
+            // Open a connection to the Edge runtime
+            ModuleClient ioTHubModuleClient = await ModuleClient.CreateFromEnvironmentAsync(settings);
```

Comment:
FXCop: All `await` statements must follow with a .ConfigureAwait(false); (in this case - since we don't care about getting back on the original context)

Reply:
Nit: here and above - use `var` if the type already exists on the same line instead of duplicating the type name.